src/sys/mod.rs

```diff
@@ -13,52 +13,81 @@
+        ) => {
+            struct $type($event_type);
 
-        impl fmt::Debug for $type {
-            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                let mut written_one = false;
-                $(
-                    $(#[$target])*
-                    #[allow(clippy::bad_bit_mask)] // Apparently some flags are zero.
-                    {
-                        // Windows doesn't use `libc` but the `afd` module.
-                        if $test(&self.0, &$libc :: $flag) {
-                            if !written_one {
-                                write!(f, "{}", stringify!($flag))?;
-                                written_one = true;
-                            } else {
-                                write!(f, "|{}", stringify!($flag))?;
+            impl fmt::Debug for $type {
+                fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                    let mut written_one = false;
+                    $(
+                        $(#[$target])*
+                        #[allow(clippy::bad_bit_mask)] // Apparently some flags are zero.
+                        {
+                            // Windows doesn't use `libc` but the `afd` module.
+                            if $test(&self.0, &$libc :: $flag) {
+                                if !written_one {
+                                    write!(f, "{}", stringify!($flag))?;
+                                    written_one = true;
+                                } else {
+                                    write!(f, "|{}", stringify!($flag))?;
+                                }
                             }
                         }
+                    )+
+                    if !written_one {
+                        write!(f, "(empty)")
+                    } else {
+                        Ok(())
                     }
-                )+
-                if !written_one {
-                    write!(f, "(empty)")
-                } else {
-                    Ok(())
                 }
             }
-        }
-    };
+        };
+    }
 }
 
 #[cfg(unix)]
-pub use self::unix::{
-    event, Event, Events, Selector, SocketAddr, SourceFd, TcpListener, TcpStream, UdpSocket,
-    UnixDatagram, UnixListener, UnixStream, Waker,
-};
+cfg_os_poll! {
+    mod unix;
+    pub use self::unix::SourceFd;
+    pub(crate) use self::unix::{event, Event, Events, Selector, Waker};
 
-#[cfg(unix)]
-mod unix;
+    cfg_tcp! {
+        pub(crate) use self::unix::{TcpListener, TcpStream};
+    }
 
-#[cfg(windows)]
-pub use self::windows::{event, Event, Events, Selector, TcpListener, TcpStream, UdpSocket, Waker};
+    cfg_udp! {
+        pub(crate) use self::unix::UdpSocket;
+    }
+
+    cfg_uds! {
+        pub(crate) use self::unix::{UnixDatagram, UnixListener, UnixStream};
+        pub use self::unix::SocketAddr;
+    }
+}
 
 #[cfg(windows)]
-mod windows;
+cfg_os_poll! {
+    mod windows;
+    pub(crate) use self::windows::*;
```

Comment:
Is there a reason for the difference with the `unix` module imports? I like this method better personally.

Reply:
Before, the `mod unix` types were re-exported as `pub use`. In this change as you've picked up on in another comment, I added `pub use(crate)`s where applicable because it helped a lot with properly structuring things visibility wise and making sure warnings appeared on things that were not used (that should have been). Unfortunately since `SourceFd` is `pub use` (and there is no equivalent in Windows), this makes the Unix module a little more explicit.

I know what you mean about preferring the glob imports here, but I really would like to keep the `pub use(crate)`s on things that should only be crate visible. It helped a lot with making sure things were not exported that should not be.